pandas/core/generic.py

```diff
@@ -2916,6 +2916,8 @@ def to_csv(self, path_or_buf=None, sep=",", na_rep='', float_format=None,
             sequence should be given if the object uses MultiIndex. If
             False do not print fields for index names. Use index_label=False
             for easier importing in R.
+        index_name: bool, default True.
```

Comment:
what is the purpose of this? we *already* have the ``index`` and ``index_label``?

Reply:
After collecting feedbacks from @gfyoung and @simonjayhawkins, if i understand correctly, a parameter `index_name` could be added into `CSVFormatter` that determines if index name should be kept/displayed or not if it is not None, since `header` in `to_csv` should only removes columns names based on its definition.

Reply:
this is what index_label already does